src/main/java/uk/gov/hmcts/probate/transformer/solicitorexecutors/ExecutorsTransformer.java

```diff
@@ -202,7 +139,26 @@ private void mapSolicitorExecutorNotApplyingListsToCaseworkerNotApplyingList(
 
     }
 
-    private  List<CollectionMember<AdditionalExecutorNotApplying>>  setExecutorNotApplyingListWithSolicitorInfo(
+    public List<CollectionMember<AdditionalExecutorApplying>> setExecutorApplyingListWithSolicitorInfo(
+            List<CollectionMember<AdditionalExecutorApplying>> execsApplying, CaseData caseData) {
+
+        // Transform list
+        if (isSolicitorExecutor(caseData) && YES.equals(caseData.getSolsSolicitorIsApplying())) {
+
+            // Add solicitor to applying list
+            execsApplying = executorListMapperService.addSolicitorToApplyingList(caseData, execsApplying);
+
+        } else if (!isSolicitorApplying(caseData)) {
```

Comment:
Do we want to remove the solicitor from the not applying list if isSolicitorExecutor is changed to NO also?

Reply:
This is a very good point, now that theyre conditional